File: app/src/main/java/com/android/androidandjava/MainActivity.java
package com.android.androidandjava;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "@@@MainActivity : ";

    private static final int RUB_IN_USD = 30;
    private static final int RUB_IN_EUR = 40;
    private static final int RUB_IN_CHF = 10;

    private EditText inputEditText = null;
    private Button conversionButton = null;
    private Button openScreenSecondButton = null;
    private TextView resultTextView = null;

    private RadioGroup currencyRadioGroup = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initViews();
        setupListeners();
    }

    private void setupListeners() {
        conversionButton.setOnClickListener(new View.OnClickListener() {
            //код ниже будет выполняться только после нажатия на кнопку.
            @Override
            public void onClick(View v) {

                //Создали переменную, присвоили ей значение текстового поля
                final String inputSrt = inputEditText.getText().toString();

                // Создали переменную, производим проверку введенного значения, в поле можно ввести только целое число
                // (если в строке будут буквы, служебные символы, компилятор выдаст ошибку. На View в поле EditText прописать
                // строку разрешающую ввод только чисел android:inputType="number",
                // минус данного решения - можно ввести только целое число)
                final double volute = parseDoubleString(inputSrt);

                //Создали переменную, присвоили ей уже обработанный в методе convert результат
                // (в метод convert положили значение текстового поля EditText)
                final double rubles = convert(volute);

                final String resultString = String.valueOf(rubles);// Создали переменную, привратили ее в строку

                resultTextView.setText(resultString);//Кладем результат в поле TextView
            }
        });

        openScreenSecondButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, SecondActivity.class);
                startActivity(intent);
                Toast.makeText(MainActivity.this, "Second Activity", Toast.LENGTH_SHORT).show();
            }
        });
    }

    //Метод проверки на формат введенного значения в поле EditText
    private double parseDoubleString(String inputSrt) {
        try {
            return Double.parseDouble(inputSrt);//Приобразовываем ткст в число
        } catch (NumberFormatException nfe) {
            return 0d;
        }
    }

    private double getCurrencyFromScreen() {
        //Вариант 1. обработка RadioGroup. наиболее предпочтителен
        switch (currencyRadioGroup.getCheckedRadioButtonId()) {
            case R.id.usa_currency_radio_button:
                return RUB_IN_USD;
            case R.id.eur_currency_radio_button:
                return RUB_IN_EUR;
            case R.id.chf_currency_radio_button:
                return RUB_IN_CHF;
            default:
                return 0;
        }
    }

    //Метод принемающий строку EditText и возващает результат (проходит вычисление)
    private double convert(double volute) {
        Log.d(TAG, "convert() called with: input = [" + volute + "]");
        return volute * getCurrencyFromScreen();
    }

    //Метод для инициализации элементов на экране (view моделе).
    private void initViews() {
        inputEditText = findViewById(R.id.input_edit_text);
        conversionButton = findViewById(R.id.result_button);
        openScreenSecondButton = findViewById(R.id.open_second_button);
        resultTextView = findViewById(R.id.result_text_view);

        currencyRadioGroup = findViewById(R.id.currency_radio_group);
    }

    @Override
    protected void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
        Log.d(TAG, "onRestoreInstanceState() called with: savedInstanceState = [" + savedInstanceState + "]");
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        Log.d(TAG, "onSaveInstanceState() called with: outState = [" + outState + "]");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d(TAG, "onStart() called");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d(TAG, "onResume() called");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d(TAG, "onPause() called");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(TAG, "onStop() called");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d(TAG, "onDestroy() called");
    }
}

File: app/src/main/java/com/android/androidandjava/MainActivity.kt
package com.android.androidandjava

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.android.androidandjava.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding //генирируется автоматически binding для всех элементов view

//    //    private lateinit var loginButton: Button
//    private val loginButton: Button by lazy { findViewById(R.id.login_button) }//вариант записи. инициализировали сразу
//    //by lazy - это линивая инициализация. lazy инициализирует первый раз до тех пор пока вы чтото не сделаете с данной переменной,
//    // например, вызовите setOnClickListener. lazy обязательно должна быть val, поэтому во фрагменте мы уже ничего не сделаем,
//    // это можно сделать только в активити.
//    //lazy обычно используют для обращения к репозеториям.
//
//    private lateinit var emailEditText: EditText
//    private lateinit var passwordEditText: EditText

    private var noteEntity: KotlinNoteEntity? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding =
            ActivityMainBinding.inflate(layoutInflater)// layoutInflater - это системный сервис андройд
        // который можно получать через контекст. это для раздувания разметки (xml файла)
        setContentView(binding.root)

        //with - это экстейшен (в данном случае это аналог apply), это способ передать binding. нет необходимости везде писать binding
        with(binding) {
            loginButton.setOnClickListener {
                val toastMassage = if (checkCredentials(
                        emailEditText.text.toString(),
                        passwordEditText.text.toString()
                    )
                ) {
                    "OK"
                } else {
                    "ERROR"
                }
                Toast.makeText(baseContext, toastMassage, Toast.LENGTH_SHORT).show()
            }
        }

        //apply - это экстейшен, это способ передать binding. нет необходимости везде писать binding
        binding.apply {
            loginButton.setOnClickListener {
                val toastMassage = if (checkCredentials(
                        emailEditText.text.toString(),
                        passwordEditText.text.toString()
                    )
                ) {
                    "OK"
                } else {
                    "ERROR"
                }
                Toast.makeText(baseContext, toastMassage, Toast.LENGTH_SHORT).show()
            }
        }

        val body = noteEntity?.body // кэширование переменной
        val title = noteEntity?.title // кэширование переменной
        if (body != null && title != null) {
            checkCredentials(body, title)
        }

        //let - это экстэншен который проверяет на нул и если не равно нулу, передает в качестве аргумента в лямду. часто используется
        noteEntity?.let {
            checkCredentials(it.body, it.title)

        }

    }

    companion object {
        fun checkCredentials(email: String, password: String): Boolean {
            return email == password
        }
    }
}